app/models/user.rb

```diff
@@ -11,18 +11,34 @@ class User < ApplicationRecord
   validates :username, presence: true
 
   # Associations
-  has_one :profile
-  has_many :requests, foreign_key: "receiver_id", dependent: :destroy
-  has_many :pending_friends, through: :requests, source: :sender
+  has_one :profile, dependent: :destroy
+
+  has_many :outgoing_requests, foreign_key: "sender_id", 
+            dependent: :destroy, class_name: 'Request'
+  has_many :outgoing_pending_friends, through: :outgoing_requests, source: :receiver
+
+  has_many :incoming_requests, foreign_key: "receiver_id", 
+            dependent: :destroy, class_name: 'Request'
+  has_many :incoming_pending_friends, through: :incoming_requests, source: :sender
 
   has_many :friendships, dependent: :destroy
-  has_many :friends, through: :friendships, source: :friend
+  has_many :friends, through: :friendships
+
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
```

Comment:
Great job here guys!

Here's another thing I would like to suggest. Can we have an instance method here in `user.rb` that allows us to list all of a user's friends? Not just those friends the user added but also those friends that added them.